prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -429,4 +447,36 @@ object CustomFHIRFunctions {
             mutableListOf(StringType(result))
         }
     }
+
+    /**
+     * Applies a timezone given by [parameters] to a date in [focus] and returns the result
+     * @return a date in the new timezone
+     */
+    fun changeTimezone(
+        focus: MutableList<Base>,
+        parameters: MutableList<MutableList<Base>>?
+    ): MutableList<Base> {
+        if (focus.size != 1) {
+            throw SchemaException("Must call changeTimezone on a single element")
+        }
+
+        val inputDate = FhirBundleUtils.convertFhirType(focus[0], focus[0].fhirType(), "dateTime") as? DateTimeType
+            ?: throw SchemaException("Must call changeTimezone on a dateTime compatible value")
+
+        if (parameters == null || parameters[0].size != 1) {
+            throw SchemaException("Must pass a timezone as the parameter")
+        }
+
+        val inputTimeZone = parameters.first().first().primitiveValue()
+        val timezonePassed = try {
+            TimeZone.getTimeZone(ZoneId.of(inputTimeZone))
+        } catch (zoneRulesException: ZoneRulesException) {
+            throw SchemaException(
+                "Invalid timezone $inputTimeZone passed. See FHIR timezone valueset " +
```

Comment:
Maybe add a link to the valueset? Also, you are using a Java ZoneId and not a FHIR value here. Perhaps change to some IEEE list or something more specific to this implementation. Do not say what ZoneId needs as that will not make sense to a user either.

Reply:
Wasn't sure about whether or not the link would be stable enough, but I suppose it probably won't change from https://hl7.org/fhir/valueset-timezones.html, will add that.

[Java ZoneId](https://docs.oracle.com/javase/8/docs/api/java/time/ZoneId.html#of-java.lang.String-) and [FHIR](https://docs.oracle.com/javase/8/docs/api/java/time/ZoneId.html#of-java.lang.String-) both use the IANA Time Zone Database for their list of time zones, and `DateTimeType`'s constuctor takes a Java TimeZone, so it should be safe to utilize Java's TimeZone and ZoneId in this implementation.